app/src/main/scala/org/alephium/explorer/Application.scala

```diff
@@ -103,14 +106,34 @@ class Application(
       _ <- mempoolSyncService.start(peers)
       _ <- tokenSupplyService.start()
       _ <- hashrateService.start()
+      _ <- finalizerService.start()
     } yield ()
   }
 
+  private def startReadWrite(): Future[Unit] = {
+    if (!readOnly) {
+      for {
+        _ <- dbInitializer.initialize()
+        _ <- if (readOnly) Future.successful(()) else startSyncService()
+      } yield ()
+
+    } else {
+      Future.successful(())
+    }
+  }
+
+  private def stopReadWrite(): Future[Unit] = {
+    if (!readOnly) {
+      blockFlowSyncService.stop()
+    } else {
+      Future.successful(())
+    }
+  }
+
   def start: Future[Unit] = {
     for {
       _       <- healthCheckDao.healthCheck()
-      _       <- dbInitializer.createTables()
-      _       <- if (readOnly) Future.successful(()) else startSyncService()
+      _       <- startReadWrite()
```

Comment:
I am not sure about the naming `startReadWrite`?

Reply:
the read-write specific stuff, like db migrations and sync services

Reply:
sure, I got what is doing by diving into the code. I meant that I don't get what the function is about by just reading the function name.

Reply:
The function itself is more like starting syncing services for write-only mode. We don't do anything for read-only mode.

Reply:
It's nitpicking of course!

Reply:
renamed in a new commit